src/test/java/org/apache/commons/crypto/random/OsCryptoRandomTest.java

```diff
@@ -48,19 +47,6 @@ public void testInvalidRandom() {
 		props.setProperty(CryptoRandomFactory.CLASSES_KEY, OsCryptoRandom.class.getName());
 		// Invalid device
 		props.setProperty(CryptoRandomFactory.DEVICE_FILE_PATH_KEY, "");
-		try {
-			CryptoRandomFactory.getCryptoRandom(props);
-			fail("Expected GeneralSecurityException");
-		} catch (final GeneralSecurityException e) {
-			Throwable cause;
-			cause = e.getCause();
-			Assert.assertEquals(IllegalArgumentException.class, cause.getClass());
-			cause = cause.getCause();
-			Assert.assertEquals(InvocationTargetException.class, cause.getClass());
-			cause = cause.getCause();
-			Assert.assertEquals(IllegalArgumentException.class, cause.getClass());
-			cause = cause.getCause();
-			Assert.assertEquals(FileNotFoundException.class, cause.getClass());
-		}
+		assertThrows(GeneralSecurityException.class, () -> CryptoRandomFactory.getCryptoRandom(props));
```

Comment:
You've changed the test here a lot, and lost a lot. I did not write this code so I can't say exactly why the test was written this way but I would not just blow away all of this.